src/Symfony/Component/Serializer/Encoder/XmlEncoder.php

```diff
@@ -292,16 +292,17 @@ private function parseXml(\DOMNode $node)
      *
      * @return array
      */
-    private function parseXmlAttributes(\DOMNode $node)
+    private function parseXmlAttributes(\DOMNode $node, array $context = array())
     {
         if (!$node->hasAttributes()) {
             return array();
         }
 
         $data = array();
+        $typeCastAttributes = $this->resolveXmlTypeCastAttributes($context);
 
         foreach ($node->attributes as $attr) {
-            if (!is_numeric($attr->nodeValue)) {
+            if (!is_numeric($attr->nodeValue) || !$typeCastAttributes) {
```

Comment:
I propose excluding number that start with zero:
if (!is_numeric($attr->nodeValue) || (isset($attr->nodeValue[1] && '0' === $attr->nodeValue[0])) {